### PR TITLE
Part of #16152: Use BoundBoxXYZ for PaintAddImageAsChild

### PR DESCRIPTION
**Pull request by frutiemax**

### CODE REVIEW

File: src/openrct2/ride/coaster/VirginiaReel.cpp [hunk 1]
```diff
@@ -186,17 +186,17 @@ void vehicle_visual_virginia_reel(
         }
     }();
     baseImage_id += ecx & 7;
-    const vehicle_boundbox* bb = &_virginiaReelBoundbox[baseImage_id >> 3];
+    const auto& vehicleBb = _virginiaReelBoundbox[baseImage_id >> 3];
+    auto bb = BoundBoxXYZ{ { vehicleBb.offset_x, vehicleBb.offset_y, vehicleBb.offset_z },
```
Comment (Gymnasiast) [inline src/openrct2/ride/coaster/VirginiaReel.cpp:190]:
```suggestion
    auto bb = BoundBoxXYZ{ { vehicleBb.offset_x, vehicleBb.offset_y, vehicleBb.offset_z + z },
```

File: src/openrct2/ride/coaster/WoodenRollerCoaster.h [hunk 2]
```diff
@@ -68,8 +68,8 @@ template<bool isClassic> void wooden_rc_track_paint_bb(PaintSession& session, co
     {
         ImageId railsImageId = wooden_rc_get_rails_colour(session).WithIndex(bb->sprite_id_b);
         PaintAddImageAsChild(
-            session, railsImageId, { bb->offset.x, bb->offset.y, height + bb->offset.z }, bb->bb_size,
-            { bb->bb_offset.x, bb->bb_offset.y, height + bb->bb_offset.z });
+            session, railsImageId, { bb->offset.x, bb->offset.y, height + bb->offset.z },
+            { { bb->bb_offset.x, bb->bb_offset.y, height + bb->bb_offset.z }, bb->bb_size });
```
Comment (Gymnasiast) [inline src/openrct2/ride/coaster/WoodenRollerCoaster.h:72]:
```suggestion
            { bb->bb_offset, bb->bb_size });
```

File: src/openrct2/ride/gentle/CarRide.cpp [hunk 3]
```diff
@@ -345,11 +345,11 @@ static void PaintCarRideStation(
     imageId = session.TrackColours[SCHEME_TRACK].WithIndex(CarRideTrackPiecesFlat[direction]);
     if (direction == 0 || direction == 2)
     {
-        PaintAddImageAsChild(session, imageId, { 0, 6, height }, { 32, 20, 1 }, { 0, 0, height });
+        PaintAddImageAsChild(session, imageId, { 0, 6, height }, { { 0, 0, height }, { 32, 20, 1 } });
     }
     else
     {
-        PaintAddImageAsChild(session, imageId, { 6, 0, height }, { 20, 32, 1 }, { 0, 0, height });
+        PaintAddImageAsChild(session, imageId, { 0, 6, height }, { { 0, 0, height }, { 20, 32, 1 } });
```
Comment (Gymnasiast) [inline src/openrct2/ride/gentle/CarRide.cpp:352]:
```suggestion
        PaintAddImageAsChild(session, imageId, { 6,,0 height }, { { 0, 0, height }, { 20, 32, 1 } });
```

File: src/openrct2/ride/gentle/MiniGolf.cpp [hunk 4]
```diff
@@ -943,22 +944,20 @@ static void PaintMiniGolfHoleC(
             break;
     }
 
+    auto bb = BoundBoxXYZ{ { boundBoxOffset.x, boundBoxOffset.y, height }, { boundBox.x, boundBox.y, 1 } };
```
Comment (Gymnasiast) [inline src/openrct2/ride/gentle/MiniGolf.cpp:947]:
```suggestion
    auto bb = BoundBoxXYZ{ { boundBoxOffset, height }, { boundBox, 1 } };
```

File: src/openrct2/ride/gentle/MiniGolf.cpp [hunk 5]
```diff
@@ -1037,22 +1036,21 @@ static void PaintMiniGolfHoleD(
...
```
Comment (Gymnasiast) [inline src/openrct2/ride/gentle/MiniGolf.cpp:1039]:
```suggestion
    auto bb = BoundBoxXYZ{ { boundBoxOffset, height }, { boundBox, 1 } };
```

File: src/openrct2/ride/gentle/MiniGolf.cpp [hunk 6]
```diff
@@ -1131,22 +1129,21 @@ static void PaintMiniGolfHoleE(
...
```
Comment (Gymnasiast) [inline src/openrct2/ride/gentle/MiniGolf.cpp:1132]:
```suggestion
    auto bb = BoundBoxXYZ{ { boundBoxOffset, height }, { boundBox, 1 } };
```

File: src/openrct2/ride/water/RiverRapids.cpp [hunk 7]
```diff
@@ -223,51 +223,43 @@ void vehicle_visual_river_rapids(
     }
     baseImage_id += carEntry->base_image_id;
 
-    const vehicle_boundbox* bb = &_riverRapidsBoundbox[j];
+    const auto& riverRapidsBb = _riverRapidsBoundbox[j];
+    auto bb = BoundBoxXYZ{ { riverRapidsBb.offset_x, riverRapidsBb.offset_y, riverRapidsBb.offset_z },
```
Comment (Gymnasiast) [inline src/openrct2/ride/water/RiverRapids.cpp:227]:
```suggestion
    auto bb = BoundBoxXYZ{ { riverRapidsBb.offset_x, riverRapidsBb.offset_y, riverRapidsBb.offset_z + z },
```

**Comment by Gymnasiast** [comment]
Please remove the additional commits and put them in another branch (preferably multiple). Reviewing just the first commit was hard enough, especially since I found several errors. I have no appetite reviewing a 1000 line PR that needs close scrutiny of every line.

**Comment by frutiemax** [comment]
Do I cherry pick with the changes from the last commit?

**Comment by frutiemax** [comment]
I've done multiple branches for the other commits, those won't build since it depends on this PR.

File: src/openrct2/ride/coaster/WoodenRollerCoaster.h [hunk 8]
```diff
@@ -68,8 +68,7 @@ template<bool isClassic> void wooden_rc_track_paint_bb(PaintSession& session, co
     {
         ImageId railsImageId = wooden_rc_get_rails_colour(session).WithIndex(bb->sprite_id_b);
         PaintAddImageAsChild(
-            session, railsImageId, { bb->offset.x, bb->offset.y, height + bb->offset.z }, bb->bb_size,
-            { bb->bb_offset.x, bb->bb_offset.y, height + bb->bb_offset.z });
+            session, railsImageId, { bb->offset.x, bb->offset.y, height + bb->offset.z }, { bb->bb_offset, bb->bb_size });
```
Comment (Gymnasiast) [inline src/openrct2/ride/coaster/WoodenRollerCoaster.h:71]:
```suggestion
            session, railsImageId, { bb->offset.x, bb->offset.y, height + bb->offset.z }, { { bb->bb_offset, height + bb->bb_offset.z }, bb->bb_size });
```

**Review by Gymnasiast — APPROVED** [review]

**Pull request merged**